projects/orquestra-workflow-shared/pyproject.toml

```diff
@@ -0,0 +1,224 @@
+[build-system]
+requires = ["hatchling", "hatch-vcs"]
+build-backend = "hatchling.build"
+
+[project]
+authors = [
+    { name = "Zapata Computing Inc.", email = "info@zapatacomputing.com" },
+]
+name = "orquestra-workflow-shared"
+description = "Compose Orquestra workflows using a Python DSL"
+readme = "README.md"
+license = { file = "LICENSE" }
+classifiers = [
+    "Programming Language :: Python :: 3",
+    "Operating System :: OS Independent",
+    "License :: OSI Approved :: Apache Software License",
+    "Topic :: Scientific/Engineering",
+]
+dynamic = ["version"]
+requires-python = ">= 3.8"
+
+# As a rule of thumb, the dependencies should be specified with lower bound
+# only.
+#
+# For pratical reasons, when choosing the lower bound, we pick the oldest
+# version we can easily install and test, even if there are older versions with
+# the same API.
+dependencies = [
+    # Schema definition
+    "pydantic>1.10.7",
+    # Pickling library
+    "cloudpickle==2.2.1",
+    # We should keep `dill` to make sure old workflows can be unpickled.
+    "dill==0.3.6",
+    # Enhances wrapping of user's functions
+    "wrapt",
+    # Configuration file locking
+    "filelock>=3.3.2",
+    # Parse version strings
+    "packaging>=21",
+    # For different import types in SDK
+    "GitPython",
+    "pip-api>=0.0.30",
+    # Python Standard Library Backports
+    "importlib_metadata>=4.0",
+    "typing-extensions>=4.1.0",
+    # Web requests for external services
+    "requests>=2.28",
+    # Nice workflow visualizations
+    "graphviz",
+    # CLI utilities
+    "argcomplete",
+    # click 8.1.4 introduced type checking errors
+    # see https://github.com/pallets/click/issues/2558
+    "click>=8.0, != 8.1.4",
+    "cloup>=2.0",
+    "inquirer>=3.0",
+    "tabulate",
+    "rich>=13.5",
+    # For automatic login
+    "aiohttp>=3.9.0",
+    # Decoding JWTs
+    "PyJWT>=2.6",
+    # Capture stdout/stderr
+    "wurlitzer>=3.0",
+    # For dremio client
+    # pyarrow 16.0 crashed ray workers for unknown reason. Crashes were not
+    # reproducable on mac - so carefull with taking that restriction away.
+    "pyarrow>=10.0,<16.0",
+    "pandas>=1.4",
+]
```

Comment:
**non-blocking**: please remember to clean up the dependencies before marking this ticket as done.

```suggestion
dependencies = [
    # Schema definition
    "pydantic>1.10.7",
    # Pickling library
    "cloudpickle==2.2.1",
    # We should keep `dill` to make sure old workflows can be unpickled.
    "dill==0.3.6",
    # Enhances wrapping of user's functions
    "wrapt",
    # Configuration file locking
    "filelock>=3.3.2",
    # Parse version strings
    "packaging>=21",
    # For different import types in SDK
    "GitPython",
    "pip-api>=0.0.30",
    # Python Standard Library Backports
    "importlib_metadata>=4.0",
    "typing-extensions>=4.1.0",
    # Web requests for external services
    "requests>=2.28",
    # Nice workflow visualizations
    "graphviz",
    # CLI utilities
    "argcomplete",
    # click 8.1.4 introduced type checking errors
    # see https://github.com/pallets/click/issues/2558
    "click>=8.0, != 8.1.4",
    "cloup>=2.0",
    "inquirer>=3.0",
    "tabulate",
    "rich>=13.5",
    # For automatic login
    "aiohttp>=3.9.0",
    # Decoding JWTs
    "PyJWT>=2.6",
    # Capture stdout/stderr
    "wurlitzer>=3.0",
    # For dremio client
    # pyarrow 16.0 crashed ray workers for unknown reason. Crashes were not
    # reproducable on mac - so carefull with taking that restriction away.
    "pyarrow>=10.0,<16.0",
    "pandas>=1.4",
]
```